mllib/src/main/scala/org/apache/spark/ml/feature/Bucketizer.scala

```diff
@@ -105,20 +106,24 @@ final class Bucketizer @Since("1.4.0") (@Since("1.4.0") override val uid: String
     transformSchema(dataset.schema)
     val (filteredDataset, keepInvalid) = {
       if (getHandleInvalid == Bucketizer.SKIP_INVALID) {
-        // "skip" NaN option is set, will filter out NaN values in the dataset
+        // "skip" NaN/NULL option is set, will filter out NaN/NULL values in the dataset
         (dataset.na.drop().toDF(), false)
       } else {
         (dataset.toDF(), getHandleInvalid == Bucketizer.KEEP_INVALID)
       }
     }
 
-    val bucketizer: UserDefinedFunction = udf { (feature: Double) =>
-      Bucketizer.binarySearchForBuckets($(splits), feature, keepInvalid)
+    val bucketizer: UserDefinedFunction = udf { (row: Row) =>
+      Bucketizer.binarySearchForBuckets(
+        $(splits),
+        row.getAs[java.lang.Double]($(inputCol)),
```

Comment:
Ideally we should use `row.getDouble(index)` and `row.isNullAt(index)` together to get values for primitive types, but technically `Row` is just a `Array[Object]`, so there is no performance penalty by using `java.lang.Double`.(this may change in the future, if possible we should prefer `isNullAt` and `getDouble`)